### PR TITLE
#211 Feature/clear topic messages

### PR DESCRIPTION
**Pull request by RamazanYapparov**
Backend part for #211

### CODE REVIEW

File: kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml [hunk 1]
```diff
@@ -309,6 +309,25 @@ paths:
...
+          schema:
+            type: string
+        - name: topicName
+          in: path
+          required: true
+          schema:
+            type: string
+      responses:
+        200:
+          description: OK
```
Comment (IldarAlmakaev) [inline kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml:331]: Please, add 404 as well

**Review by IldarAlmakaev — COMMENTED** [review]
Left a minor comment for the API contract.
Overall, looks good to me
Good job!

**Review by IldarAlmakaev — APPROVED** [review]

File: kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml (hunk 1 above)
Comment (RamazanYapparov) [reply kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml:333]: Added, thanks

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/cluster/service/ClusterService.java [hunk 2]
```diff
@@ -196,4 +196,18 @@ private KafkaCluster updateCluster(String topicToDelete, String clusterName, Kaf
                 .map(c -> consumingService.loadMessages(c, topicName, consumerPosition, query, limit))
                 .orElse(Flux.empty());
     }
+
+    public Mono<Void> deleteTopicMessages(String clusterName, String topicName, Optional<Integer> partition) {
```
Comment (germanosin) [inline kafka-ui-api/src/main/java/com/provectus/kafka/ui/cluster/service/ClusterService.java:200]: maybe partition list?

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/cluster/service/ConsumingService.java [hunk 3]
```diff
@@ -55,6 +57,12 @@
 				.limitRequest(recordsLimit);
 	}
 
+	public Mono<Map<TopicPartition, Long>> loadOffsets(KafkaCluster cluster, List<TopicPartition> partitions) {
```
Comment (germanosin) [inline kafka-ui-api/src/main/java/com/provectus/kafka/ui/cluster/service/ConsumingService.java:60]: we could do it simpler) just create consumer and get offsets.
Btw offset could be -1 for empty partitions

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/kafka/KafkaService.java [hunk 4]
```diff
@@ -490,4 +492,10 @@ private InternalClusterMetrics calculateClusterMetrics(InternalClusterMetrics in
             return Collections.emptyMap();
         }
     }
+
+    public Mono<Void> deleteTopicMessages(KafkaCluster cluster, String topicName, Map<TopicPartition, Long> offsets) {
+        var records = offsets.entrySet().stream().map(entry -> Map.entry(entry.getKey(), RecordsToDelete.beforeOffset(entry.getValue()))).collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
```
Comment (germanosin) [inline kafka-ui-api/src/main/java/com/provectus/kafka/ui/kafka/KafkaService.java:497]: please wrap into multiple lines

File: kafka-ui-api/src/test/java/com/provectus/kafka/ui/producer/KafkaTestProducer.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        return new KafkaTestProducer<>(new KafkaProducer<>(Map.of(
+                ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, kafkaContainer.getBootstrapServers(),
+                ProducerConfig.CLIENT_ID_CONFIG, "KafkaTestProducer",
+                ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class,
+                ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class
+        )));
+    }
+
+    public void send(String topic, ValueT value) {
+        producer.send(new ProducerRecord<>(topic, value));
```
Comment (germanosin) [inline kafka-ui-api/src/test/java/com/provectus/kafka/ui/producer/KafkaTestProducer.java:28]: what about key?

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/cluster/service/ClusterService.java (hunk 2 above)
Comment (RamazanYapparov) [reply kafka-ui-api/src/main/java/com/provectus/kafka/ui/cluster/service/ClusterService.java]: Fixed

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/cluster/service/ConsumingService.java (hunk 3 above)
Comment (RamazanYapparov) [reply kafka-ui-api/src/main/java/com/provectus/kafka/ui/cluster/service/ConsumingService.java]: Thanks, fixed

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/kafka/KafkaService.java (hunk 4 above)
Comment (RamazanYapparov) [reply kafka-ui-api/src/main/java/com/provectus/kafka/ui/kafka/KafkaService.java]: Done

**Review by germanosin — APPROVED** [review]

**Pull request merged**